qiita_db/processing_job.py

```diff
@@ -1196,7 +1208,7 @@ def _raise_if_not_in_construction(self):
                      WHERE processing_job_workflow_id = %s"""
             qdb.sql_connection.TRN.add(sql, [self.id])
             res = qdb.sql_connection.TRN.execute_fetchflatten()
-            if len(res) != 1 or res[0] != 'in_construction':
+            if (len(res) == 1 and res[0] != 'in_construction') or len(res) > 1:
```

Comment:
Could you add some documentation of each test? I'm not following why/when len(res) > 1 ...